### PR TITLE
If run_if throws exception (not intentionally), test will terminate with pass outcome

### PR DESCRIPTION
**Pull request by eli-zr**
More details are provided inside [issue](https://github.com/google/openhtf/issues/1202)

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/openhtf/1203)

### CODE REVIEW

File: openhtf/core/phase_executor.py [hunk 1]
```diff
@@ -307,8 +307,10 @@ def _execute_phase_once(
       try:
         run_phase = phase_desc.options.run_if()
       except Exception:  # pylint: disable=broad-except
+        self.logger.debug('Phase %s stopped due to a fault in run_if function.',
+                        phase_desc.name)
```
Comment (glados-verma) [inline openhtf/core/phase_executor.py:311]: Please line up the start of this line with the opening parenthesis on the previous line. Make sure you're using spaces and not tabs to do the alignment.

File: openhtf/core/phase_executor.py [hunk 2]
```diff
@@ -303,10 +303,18 @@ def _execute_phase_once(
...
-      self.logger.debug('Phase %s skipped due to run_if returning falsey.',
+    if phase_desc.options.run_if:
+      try:
+        run_phase = phase_desc.options.run_if()
+      except Exception:  # pylint: disable=broad-except
+        # Allow graceful termination
+        return PhaseExecutionOutcome(phase_descriptor.PhaseResult.STOP), None
+
+      if not run_phase:
+        self.logger.debug('Phase %s skipped due to run_if returning falsey.',
```
Comment (glados-verma) [inline openhtf/core/phase_executor.py:316]: Same as a previous comment, please line up the next line with the opening parenthesis.

File: test/core/phase_executor_test.py [hunk 3]
```diff
@@ -54,3 +55,22 @@ def test_execute_phase_with_repeat_limit_max_exceeds_default_limit(self):
...
+
+    _tr = None
+    def final(tr):
+      nonlocal _tr
+      _tr = tr
+
+    test = openhtf.Test(phase)
+    test.add_output_callbacks(final)
+    test.execute()
+    self.assertEqual(_tr.outcome, test_record.Outcome.ERROR)
```
Comment (glados-verma) [inline test/core/phase_executor_test.py:76]: I suggest switching from unittest.TestCase to (openhtf.util) test.TestCase, and then you can use `execute_phase_or_test` and the appropriate `assertTest` method.

**Review by glados-verma — CHANGES_REQUESTED** [review]

File: openhtf/core/phase_executor.py (hunk 2 above)
Comment (eli-zr) [reply openhtf/core/phase_executor.py:316]: Done

File: test/core/phase_executor_test.py (hunk 3 above)
Comment (eli-zr) [reply test/core/phase_executor_test.py]: Done

**Review by glados-verma — APPROVED** [review]

**Pull request merged**